trace.go

```diff
@@ -0,0 +1,25 @@
+package middleware
+
+import (
+	"context"
+
+	"go.opencensus.io/trace"
+)
+
+type Tracer interface {
+	StartSpan(ctx context.Context, resourceName string) (context.Context, Span)
+}
+
+type Span interface {
+	End()
+}
+
+type OpenCensusTracer struct{}
+
+func (t *OpenCensusTracer) StartSpan(ctx context.Context, resourceName string) (context.Context, Span) {
+	if trace.FromContext(ctx) == nil {
+		return ctx, nil
```

Comment:
Shouldn't we start a new root span if there's no span in the current context? It will look less pretty in the flame graph, but at least we get something if no earlier middleware has created a root span and attached it to the context.

Reply:
The behavior is ported from the old middleware. It is needed if you have the health endpoint in the same webserver. Added a comment to explain.